File: python/netcdf_info.py

#!/usr/bin/env python

#  Author: Laur Ferris
#  Email address: lnferris@alum.mit.edu
#  Website: https://github.com/lnferris/ocean_data_tools
#  Jun 2020; Last revision: 16-Jun-2020
#  Distributed under the terms of the MIT License

import netCDF4

# Call the dataset constructor for one of the data files.
filename = '/Users/lnferris/Desktop/33RO20150525/33RO20150525_00113_00001_ctd.nc'
dset = netCDF4.Dataset(filename)

# Print information about the variables in the file.
print(dset.variables)


# Sample output.....

#   (u'woce_date', <type 'netCDF4._netCDF4.Variable'>
#   int32 woce_date(time)
#        long_name: WOCE date
#        units: yyyymmdd UTC
#        data_min: 20150527
#       data_max: 20150527
#        C_format: %8d
#   unlimited dimensions:
#   current shape = (1,)

File: utilities/netcdf_info.m
%  Author: Laur Ferris
%  Email address: lnferris@alum.mit.edu
%  Website: https://github.com/lnferris/ocean_data_tools
%  Jun 2020; Last revision: 25-Jun-2020
%  Distributed under the terms of the MIT License

function netcdf_info(nc_dir)

full_path = dir(nc_dir);

for i = 1:length(full_path) % For each file in full_path...
    
    try
        delete netcdf_info-ocean_data_tools
        diary netcdf_info-ocean_data_tools
        ncdisp([full_path(i).folder '/' full_path(i).name])
        diary off
        disp(['Info about ',full_path(i).name,' saved to ', pwd,'/netcdf_info-ocean_data_tools.'])
        break
    catch
    end

end

